src/collections.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * A {@link ReadonlyMap} that cannot be modified after instantiation.
+ * The implementation uses an inner map hidden via a private field, and the
+ * immutability guarantee relies on it being impossible to get a reference
+ * to this map.
+ */
+class FrozenMap<Key, Value> implements ReadonlyMap<Key, Value> {
+  readonly #map: Map<Key, Value>;
+
+  public get size() {
+    return this.#map.size;
+  }
+
+  public [Symbol.iterator]() {
+    return this.#map[Symbol.iterator]();
+  }
+
+  constructor(entries?: readonly (readonly [Key, Value])[] | null) {
```

Comment:
Why the double `readonly`? Just curious.

Reply:
What do you think about putting the `constructor` above `get size` and the iterator method?

Reply:
On mobile right now, but IIRC `readonly` isn't recursive, so the double was necessary.

Regarding the place of getters etc., we've been putting properties before the construction by convention (or necessity?), and getters seem more conceptually similar to properties than methods. Class "item" (what's the word for properties and methods?) order is not something we've ever standardized, although it would be cool if we had a lint rule that imposed some kind of order.

Reply:
I'm pretty agnostic about where the properties go, but I merged this so as to not lost the approvals. Maybe we can create an issue on the lint config?